### ISSUE TITLE
error: '__class__' is not defined

### ISSUE THREAD

**Issue by williamfzc** [microsoft/pyright#269]
**Describe the bug**
`'__class__' is not defined` was reported by pyright when I try to call `__class__` within a Class.

**To Reproduce**
Every time you call `__class__`.

**Expected behavior**
Actually I do not know what the best way is.

**Screenshots or Code**

you can try this:

```python
class A:
    def __init__(self, a):
        self.a = a

    def hello(self) -> "A":
        return __class__(456)


some = A(123)
hello_res = some.hello()
print(hello_res.a)
```

pyright reported ` 6:16 - error: '__class__' is not defined`, and if you:

```python
class A:
    def __init__(self, a):
        self.a = a

    def hello(self) -> "A":
        return self.__class__(456)


some = A(123)
hello_res = some.hello()
print(hello_res.a)
```

pyright reported nothing. but PyCharm said that `Expected type A, got Optional[Type[A]] instead`.

**VS Code extension or command-line**
1.0.66

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. You're correct — pyright wasn't properly handling the implicit `__class__` name that the python interpreter automatically adds to methods if they access `__class__` or `super`. I've fixed the issue, and it will be in the next version of pyright.

**Comment by erictraut** [comment]
This is now fixed in version 1.0.67, which I just published.
